File: jdba-core-dba/src/main/java/org/jetbrains/jdba/core/BaseCommandRunner.java
package org.jetbrains.jdba.core;

import org.jetbrains.jdba.intermediate.IntegralIntermediateSeance;



/**
 * @author Leonid Bushuev from JetBrains
 */
public class BaseCommandRunner implements DBCommandRunner {

  private final IntegralIntermediateSeance myInterSeance;

  protected BaseCommandRunner(final IntegralIntermediateSeance interSeance) {
    myInterSeance = interSeance;
  }

  @Override
  public DBCommandRunner withParams(final Object... params) {
    myInterSeance.setInParameters(params);
    return this;
  }

  @Override
  public DBCommandRunner run() {
    myInterSeance.execute();
    return this;
  }
}

File: DekafMain/src/impl/BaseCommandRunner.kt
package org.jetbrains.dekaf.impl

import org.jetbrains.dekaf.core.DBCommandRunner
import org.jetbrains.dekaf.core.TaskKind.TASK_COMMAND
import org.jetbrains.dekaf.inter.InterSeance
import org.jetbrains.dekaf.inter.InterTask


internal class BaseCommandRunner: BaseRunner, DBCommandRunner {

    constructor(session: BaseSession, inter: InterSeance, text: String) : super(session, inter, text)

    override fun prepare() {
        val task = InterTask(TASK_COMMAND, text)
        inter.prepare(task)
    }

    override fun withParams(vararg params: Any?): BaseCommandRunner {
        super.withParams(*params)
        return this
    }

    override fun run(): BaseCommandRunner {
        inter.execute()
        return this
    }

    override val affectedRowsCount: Int
        get() = inter.affectedRowsCount
}